<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CodingWidget</class>
 <widget class="QWidget" name="CodingWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>725</width>
    <height>399</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>10</number>
   </property>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-1649</y>
        <width>707</width>
        <height>2142</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label">
         <property name="font">
          <font>
           <pointsize>12</pointsize>
          </font>
         </property>
         <property name="lineWidth">
          <number>2</number>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;We will like to know just how technical you are. &lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;Please Answer the following Short Questions in order to successfully &lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;save your profile&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="lblQ1">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Question 1&lt;/span&gt; - In the following code, what will be output if you compile and execute the following code?&lt;br/&gt;&lt;br/&gt;&lt;span style=&quot; color:#ffa500;&quot;&gt;int&lt;/span&gt; main (&lt;span style=&quot; color:#ffa500;&quot;&gt;int&lt;/span&gt; argc, &lt;span style=&quot; color:#ffa500;&quot;&gt;char&lt;/span&gt; *argv[]) {&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#ffa500;&quot;&gt;int&lt;/span&gt; i = 320;&lt;br/&gt;&lt;span style=&quot; color:#ffa500;&quot;&gt;int&lt;/span&gt; *b = &amp;amp;i;&lt;br/&gt;&lt;span style=&quot; color:#ffa500;&quot;&gt;int&lt;/span&gt; **ptr = &amp;amp;b;&lt;br/&gt;    cout &amp;lt;&amp;lt; *ptr&lt;/p&gt;&lt;p&gt;}&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q1Opt1">
         <property name="text">
          <string>320</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q1ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q1Opt2">
         <property name="text">
          <string>Address of i</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q1ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q1Opt3">
         <property name="text">
          <string>Address of b</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q1ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q1Opt4">
         <property name="text">
          <string>Compiler Error</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q1ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="lblQ2">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Question 2&lt;/span&gt; - In a Deque data structure , the only ways you can insert and remove information is with:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q2Opt1">
         <property name="text">
          <string>pushfront(), popback()</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q2ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q2Opt2">
         <property name="text">
          <string>pushfront(), popback(), popfront(), pushback()</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q2ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q2Opt3">
         <property name="text">
          <string>pushfront(), popfront()</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q2ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q2Opt4">
         <property name="text">
          <string>pushback(), popback()</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q2ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="lblQ3">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Question 3&lt;/span&gt; - In the following code, what will is the value of c if you compile and execute the following code (Note ^ is the operator for XOR)?&lt;br/&gt;&lt;br/&gt;&lt;span style=&quot; color:#ffa500;&quot;&gt;int&lt;/span&gt; main (&lt;span style=&quot; color:#ffa500;&quot;&gt;int&lt;/span&gt; argc, &lt;span style=&quot; color:#ffa500;&quot;&gt;char&lt;/span&gt; *argv[]) {&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#ffa500;&quot;&gt;unsignedchar&lt;/span&gt; a = 0b01000000;&lt;br/&gt;&lt;span style=&quot; color:#ffa500;&quot;&gt;unsignedchar&lt;/span&gt; b = 0b01010111;&lt;br/&gt;&lt;span style=&quot; color:#ffa500;&quot;&gt;unsignedchar&lt;/span&gt; c = a^b;&lt;/p&gt;&lt;p&gt;}&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q3Opt1">
         <property name="text">
          <string>0b00010111</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q3ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q3Opt2">
         <property name="text">
          <string>0b01010111</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q3ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q3Opt3">
         <property name="text">
          <string>0b01000000</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q3ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q3Opt4">
         <property name="text">
          <string>0b01111111</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q3ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="lblQ4">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Question 4&lt;/span&gt; - What is the purpose of extern?&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q4Opt1">
         <property name="text">
          <string>A storage specifier that resolves the scope of global symbols.</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q4ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q4Opt2">
         <property name="text">
          <string>It is the used in the syntax for deriving classes</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q4ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q4Opt3">
         <property name="text">
          <string>It is used to specify a class in which it can access all the private attributes</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q4ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q4Opt4">
         <property name="text">
          <string>This term does not exist in C++</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q4ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="lblQ5">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Question 5 &lt;/span&gt;- What is function overloading?&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q5Opt1">
         <property name="text">
          <string>Calling a funtion from another function</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q5ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q5Opt2">
         <property name="text">
          <string>Having more than one function of same name</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q5ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q5Opt3">
         <property name="text">
          <string>A function without parameters</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q5ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="q5Opt4">
         <property name="text">
          <string>When a function calls itself</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">q5ButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="lblQ6">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Question 6&lt;/span&gt;:&lt;/p&gt;&lt;p&gt;Write a function that produces the max three numbers in a list of integers.&lt;/p&gt;&lt;p&gt;The Function description and prototype is as follows:&lt;/p&gt;&lt;p&gt;/*!&lt;/p&gt;&lt;p&gt;* &lt;span style=&quot; color:#00aa00;&quot;&gt;@param&lt;/span&gt;: (in) listToCheck: &lt;span style=&quot; color:#ffa500;&quot;&gt;int*&lt;/span&gt;, &lt;/p&gt;&lt;p&gt;* --------------(in) numElementsInList: &lt;span style=&quot; color:#ffa500;&quot;&gt;int&lt;/span&gt;, &lt;/p&gt;&lt;p&gt;* --------------(out) maxThreeElementsInList: &lt;span style=&quot; color:#ffa500;&quot;&gt;int*&lt;/span&gt;&lt;/p&gt;&lt;p&gt;* &lt;span style=&quot; color:#00aa00;&quot;&gt;@description&lt;/span&gt;: This function takes a integer array and a count, and an &lt;/p&gt;&lt;p&gt;* ----------------------empty arrary to populate with the maximum three&lt;/p&gt;&lt;p&gt;* ----------------------elements in the list.&lt;/p&gt;&lt;p&gt;* &lt;span style=&quot; color:#00aa00;&quot;&gt;@output&lt;/span&gt;: &lt;span style=&quot; color:#ffa500;&quot;&gt;void&lt;/span&gt;&lt;/p&gt;&lt;p&gt;*/&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#ffa500;&quot;&gt;void&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt; maximumThreeNumbersInList(&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ffa500;&quot;&gt;int* &lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;list, &lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ffa500;&quot;&gt;int&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt; count, &lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ffa500;&quot;&gt;int* &lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;maxThree);&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Please Write your implementation for the function in the text field below.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Ensure that your function prototype matches the one above:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="txtEdit">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>400</height>
          </size>
         </property>
         <property name="font">
          <font>
           <pointsize>12</pointsize>
          </font>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="q1ButtonGroup"/>
  <buttongroup name="q2ButtonGroup"/>
  <buttongroup name="q3ButtonGroup"/>
  <buttongroup name="q4ButtonGroup"/>
  <buttongroup name="q5ButtonGroup"/>
 </buttongroups>
</ui>
